pandas/tests/resample/test_timedelta.py

```diff
@@ -114,14 +115,56 @@ def test_resample_timedelta_values():
     # check that timedelta dtype is preserved when NaT values are
     # introduced by the resampling
 
-    times = timedelta_range("1 day", "4 day", freq="4D")
+    times = timedelta_range("1 day", "6 day", freq="4D")
     df = DataFrame({"time": times}, index=times)
 
-    times2 = timedelta_range("1 day", "4 day", freq="2D")
+    times2 = timedelta_range("1 day", "6 day", freq="2D")
     exp = Series(times2, index=times2, name="time")
     exp.iloc[1] = pd.NaT
 
     res = df.resample("2D").first()["time"]
     tm.assert_series_equal(res, exp)
     res = df["time"].resample("2D").first()
     tm.assert_series_equal(res, exp)
+
+
+@pytest.mark.parametrize(
+    "start, end, freq, resample_freq",
+    [
+        ("8H", "21h59min", "10S", "3H"),
```

Comment:
Might be useful to include some of the cases that worked well before the PR? (see #30353)
```suggestion
        ("0H", "21h59min50s", "10S", "3H"),
        ("8H", "21h59min50s", "10S", "2H"),
```

Reply:
I added this in my last commit along with some other working example for the other test and this one. They all worked before this PR (I checked on master). :wink: